Handler/Events.hs

```diff
@@ -58,14 +58,26 @@ addListMetaData urlRender totalCount keyValues =
             ]
 
 
+textToOrder :: Text -> SelectOpt Event
+textToOrder text =
+    case textWithNoPrefix of
+        "id"    -> direction text $ EventId
+        "title" -> direction text $ EventTitle
+        "user"  -> direction text $ EventUserId
+        _       -> error "Wrong order"
+
+    where textWithNoPrefix = if T.isPrefixOf "-" text
+                then T.tail text
+                else text
+          direction t = if T.isPrefixOf "-" t
```

Comment:
@bitamar the "trick" here is to pass the text as argument (i.e. `direction t`) - so it's inferred correctly by the compiler. Otherwise, there's also the explicit type hinting:

```haskell
direction :: forall record typ. Text 
          -> EntityField record typ 
          -> SelectOpt record
direction t = if T.isPrefixOf "-" t
          then Desc
          else Asc
```